trieste/models/gpflow/models.py

```diff
@@ -328,17 +362,17 @@ def conditional_predict_f(
         if isinstance(self.model, SGPR):
             raise NotImplementedError("Conditional predict f is not supported for SGPR.")
 
-        mean_add, cov_add = self.model.predict_f(
-            additional_data.query_points, full_cov=True
+        mean_add, cov_add = self.predict_joint(
+            additional_data.query_points
```

Comment:
maybe stress why we dont use the posterior object here (i.e. this is model fitting)